edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -90,40 +94,56 @@ public void ComputeAvailability(ModuleSet desired, ModuleSet current)
             /* Add new modules to track */
             foreach (string module in down.Union(up))
             {
-                this.availabilities.Add(new Availability(module, this.time));
+                this.availabilities.Add(new Availability(module, this.systemTime));
             }
         }
 
-        TimeSpan CalculateEdgeAgentDowntime()
+        public void Dispose()
+        {
+            this.updateCheckpointFile.Dispose();
+        }
+
+        /*
+         * The Logic below handles edgeAgent's own avaliability. It keeps a checkpoint file containing the current timestamp
+         *  that it updates every 5 minutes. On a clean shutdown, it deletes this file to indicate it shouldn't be running. If agent crashes
+         *  or returns a non-zero code, it leaves the file. On startup, if the file exists, agent knows it shutdown incorrectly \
+         *  and can calculate its downtime using the timestamp in the file.
+         */
+        public void IndicateCleanShutdown()
         {
-            AppDomain.CurrentDomain.ProcessExit += this.NoteCurrentTime;
             try
             {
-                if (File.Exists("shutdown_time"))
-                {
-                    // TODO: get iotedged uptime. if < a couple minutes, assume intentional shutdown and return 0.
-                    long ticks = long.Parse(File.ReadAllText("shutdown_time"));
-                    return this.time.UtcNow - new DateTime(ticks);
-                }
+                File.Delete(this.checkpointFile);
             }
             catch (Exception ex)
             {
-                this.log.LogError($"Could not load shutdown time:\n{ex}");
+                this.log.LogError($"Could not delete checkpoint file:\n{ex}");
```

Comment:
Should we be using https://docs.microsoft.com/en-us/dotnet/api/system.environment.newline?view=netframework-4.8 instead of /n ?

Reply:
we use \n everywhere else in agent and hub.

Reply:
Doesn't the logger have a method overload that takes an exception directly?